components/resolved_ts/src/advance.rs

```diff
@@ -396,10 +386,16 @@ impl LeadershipResolver {
             match res {
                 Ok((to_store, resp)) => {
                     for region_id in resp.regions {
-                        resp_map
-                            .entry(region_id)
-                            .or_insert_with(|| Vec::with_capacity(store_count))
-                            .push(to_store);
+                        if let Some(prog) = progresses.get_mut(&region_id) {
+                            if prog.resolved {
```

Comment:
can we only check region_id in `valid_regions` here and remove resolved?

Reply:
Check whether a `region_id` is in `valid_regions` or not utilizes more CPU usage. It may be a problem, although I'm not sure.